### PR TITLE
Audit: Make email and name checks case-insensitive by lowercasing all

### LINKED ISSUE #47: Email and Name checks should not be case sensitive
Email addresses still resolve no matter the case used and a person is still the same person if their name is in all caps or all lower case. This would also prevent false failures if someone accidentally capitalizes the first 2 letters in their name or something when typing.

### PR DESCRIPTION
**Pull request by jzaefferer**
Changes the signatures module to return lower cased email and name properties, and updates the audit to lower case commit email and name.

Fixes #47

### CODE REVIEW

File: lib/repo/audit.js [hunk 1]
```diff
@@ -41,20 +41,22 @@ Repo.prototype.audit = function( options ) {
...
 			}
 
 			// At least one name must match
-			var noMatchingNames = signatures[ commit.email ].names.indexOf( commit.name ) === -1;
+			var noMatchingNames = signatures[ email ].names.indexOf( name ) === -1;
 			if ( noMatchingNames ) {
-				signatures[ commit.email ].errors.push(
-					"Name on signature (" + signatures[ commit.email ].names.join( ", " ) +
+				signatures[ email ].errors.push(
+					"Name on signature (" + signatures[ email ].names.join( ", " ) +
```
Comment (scottgonzalez) [inline lib/repo/audit.js]: I would expect this message to keep the original casing, even if the comparison ignores casing.

File: lib/signatures.js [hunk 2]
```diff
@@ -81,8 +81,8 @@ function validate( entries ) {
 
 		// Map to name and email
 		return {
-			names: [ unorm.nfc( row.gsx$fullname.$t ) ],
-			email: row.gsx$emailaddress.$t,
+			names: [ unorm.nfc( row.gsx$fullname.$t.toLowerCase() ) ],
```
Comment (jzaefferer) [inline lib/signatures.js]: I'll drop this `toLowerCase()` and modify the `noMatchingNames` check to be case-insensitive.

**Comment by jzaefferer** [comment]
Updated to keep names intact and only lowercase them for the comparison.

**Comment by scottgonzalez** [comment]
Looks good.

**Pull request merged**